### PR TITLE
Initial README.md

### PR DESCRIPTION
**Pull request by MMarco94**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,24 @@
...
+# Couch-Tracker
+
+An Android application, to keep track of the TV shows you follow, and the movies you've watched.
+
+This repository contains all the documents and design decisions we've made.
+
+
+## Tenets
+
+- Free and open-source
```
Comment (MMauro94) [inline README.md]: Do we want to think about where the publishing will take place? Just GitHub Releases? FDroid? Play Store?

File: README.md [hunk 2]
```diff
@@ -1 +1,24 @@
...
+- No ads, trackers, subscriptions, etc.
```
Comment (MMauro94) [inline README.md]: For the subscriptions part, I would add: "for anything that the app does locally". If, in the far future, we want to create a cloud-based service to keep your data, for example, that could be subscription-based.

Maybe we can split this tenent into two:
- no ads, no trackers, no third party integrations which are not strictly necessary to make the app work
- no paywalling features that the app does locally and that do not require an ongoing cost from us

File: README.md (hunk 1 above)
Comment (MMarco94) [reply README.md]: Seems premature

File: README.md [hunk 3]
```diff
@@ -1 +1,24 @@
...
+- Maintaining and developing the app can be an hobby: low (or zero) fees, low architectural complexity, supported by the community
```
Comment (MMauro94) [inline README.md]: > can be an hobby

I don't know how much sense it makes to make this statement here. Agree on the "low (or zero) fees, low architectural complexity", but the hobby part should be just for us. Tenents of the app is something user should be able to read and understand, this is out of scope for a public doc. Also, anything free and open-source by design is not set up to be supported like a paid service, this doesn't need to be said.

Maybe we can rephrase it to something like this:
- Low technical complexity: as much client-side code as possible, there should be no dependency on us keeping it up for it to work

On this note: we also need to decide what to do with the sadly required external service (TMDB): do users need to bring their own API keys? Or is shipping our own API key a compromise we're willing to make?

File: README.md (hunk 2 above)
Comment (MMarco94) [reply README.md]: Updated

File: README.md [hunk 4]
```diff
@@ -1 +1,24 @@
...
+- User data is owned by and hosted by the respective user
```
Comment (MMauro94) [inline README.md]: "Hosted" is the wrong word here. I think "owned" is enough. Can we change it to:

- User data is owned by each user, and is always available to them. This data is in an open format.

File: README.md (hunk 3 above)
Comment (MMarco94) [reply README.md]: Updated

File: README.md [hunk 5]
```diff
@@ -1 +1,24 @@
...
+- Fast and simple
+
```
Comment (MMauro94) [inline README.md:17]: Can we also add:

- Focused on user data: discover/explore sections and similar stuff may have a place, but it is not central

File: README.md (hunk 4 above)
Comment (MMarco94) [reply README.md]: Updated

File: README.md (hunk 5 above)
Comment (MMarco94) [reply README.md:17]: Added

File: README.md [hunk 6]
```diff
@@ -1 +1,24 @@
...
+
+## General overview
+
+The app stores the user data in a file, logically owned by the user. The file will be in the user partition, and could be synced with other devices (e.g. using Synthing), backed up in the cloud, or read/written by other apps that understand the same file format.
```
Comment (MMauro94) [inline README.md]: > The file will be in the user partition

I agree that the app must handle this case, but we also might think of offering a more "standard" solution where the user data is in the normal app data storage and is synced by the Play Store, like most apps work. In this case there must be a button that allows you to export the file as is.

This would remove attrition for non-techincal users to use it.

File: README.md [hunk 7]
```diff
@@ -1 +1,24 @@
...
+
+All the shows/movies/episodes/collections/actors referenced by the last open user data file will be downloaded and kept in sync automatically, this time in a local database owned by the app.
+This local data will be kept for a brief period of time after it is detected as unused, so switching between different user data files won't cause a complete re-download of everything.
+
+Notifications, widgets, and all other aspects of the app that don't require active interactions will reference the last open user data file.
```
Comment (MMauro94) [inline README.md]: I don't like this "last open user data file" at all. I think we can either:
- explicitly make the app single db. There is no list of "recent" stuff open etc, simply the app assumes there is on db and one db only, which is what you set up on first install. If you go deep into the settings you can "logout" and start over with a new file, but I wouldn't make it easy if we don't support multi-db in a deeper way.
- have some sort of multi-db support where you can "add" dbs like you would add accounts on some other similar apps. This way the app can keep track of what is what and keep data/generate notifications and so on based on that.

In all honesty I think we can go with the first one, I don't see many use-cases where someone would need to have more than one db. On shared devices, Android already supports multiple accounts that gives each app its own data dir.

File: README.md (hunk 6 above)
Comment (MMarco94) [reply README.md]: ok

File: README.md (hunk 7 above)
Comment (MMarco94) [reply README.md]: Ok, let's go with option 1

File: README.md (hunk 3 above)
Comment (MMarco94) [reply README.md]: I'd say binary releases could come with default API keys. Building the app will require a key (either at build time, or the user will need to enter it explicitly in the app's settings).

Out of scope of this section though. We can add a specific section about external data.

File: README.md [hunk 8]
```diff
@@ -1 +1,25 @@
...
+
+## Tenets
+
+- Open-source
+- no ads, no trackers and no third party integrations which are not strictly necessary to make the app work
+- no paywalling for features done locally and don't require an ongoing cost from us
+- Low (or zero) licensing/hosting fees, low architectural complexity
+- User data is owned by each user, and is always available to them. This data is in an open format
+- Focused on user data: discover/explore sections and the likes may have a place, but it is not central
+- Fast and simple
```
Comment (MMauro94) [inline README.md]: I think we can put this as a second point

File: README.md [hunk 9]
```diff
@@ -1 +1,25 @@
...
+# Couch-Tracker
+
+An Android application, to keep track of the TV shows you follow, and the movies you've watched.
+
+This repository contains all the documents and design decisions we've made.
+
+
+## Tenets
+
+- Open-source
```
Comment (MMauro94) [inline README.md:10]: Why no more free? 😂

File: README.md [hunk 10]
```diff
@@ -1 +1,25 @@
...
+- no ads, no trackers and no third party integrations which are not strictly necessary to make the app work
+- no paywalling for features done locally and don't require an ongoing cost from us
```
Comment (MMauro94) [inline README.md]: and **that** don't require

**Comment by MMauro94** [comment]
Nitpick: can you have each point starting with an uppercase letter?

File: README.md [hunk 11]
```diff
@@ -1 +1,25 @@
...
+- Low (or zero) licensing/hosting fees, low architectural complexity
```
Comment (MMauro94) [inline README.md]: I this is a bit odd read from a user perspective, "as much client-side code as possible, no dependency on us" is a bit more clear

File: README.md (hunk 8 above)
Comment (MMarco94) [reply README.md]: ok

File: README.md (hunk 9 above)
Comment (MMarco94) [reply README.md:10]: Well, the "no paywalling for features done locally and don't require an ongoing cost from us" kind of covers that.

Also, if we want to keep the door open for charging users when/if we build a managed cloud service, I'd rather be upfront now.

Comment (MMauro94) [reply README.md:10]: ah ok, makes sense

**Review by MMauro94 — APPROVED** [review]
LGTM. Please squash the commits when you merge
